### PR TITLE
Rockfield: Fix the list font - applied via Global Styles.

### LINKED ISSUE #2526: Rockfield: Headings font applied to list items
## Steps to replicate

1. Add a list block, a paragraph, and heading block(for comparison) to a post or a page.
2. Go to global styles and change body font, then hit Publish.
3. In the editor the font changes correctly in both the paragraph and the list.
4. Preview the post. Notice that the list is using the headings font, not the body font.

I noticed this happens in Rockfield and Mayland (and potentially others) - both originally using the same font for body and headings. 

image

image

### PR DESCRIPTION
**Pull request by danieldudzic**
Closes #2526

### CODE REVIEW

File: rockfield/sass/_config-child-theme-deep.scss [hunk 1]
```diff
@@ -266,8 +266,8 @@ $config-list: (
 	// Fonts
 	"font": (
 		"family": (
-			"fallback": map-deep-get($config-global, "font", "family", "primary", "fallback"),
-			"css-var": map-deep-get($config-global, "font", "family", "primary", "css-var"),
+			"fallback": map-deep-get($config-global, "font", "family", "secondary", "fallback"),
+			"css-var": map-deep-get($config-global, "font", "family", "secondary", "css-var"),
```
Comment (scruffian) [inline rockfield/sass/_config-child-theme-deep.scss]: Are these lines actually used?

Comment (danieldudzic) [reply rockfield/sass/_config-child-theme-deep.scss]: Umm, yes. They compile to:


```
ul,
ol {
	font-family: Lora, Baskerville, Georgia, Times, serif;
	font-family: var(--font-base, Lora, Baskerville, Georgia, Times, serif);
}
```

Comment (danieldudzic) [reply rockfield/sass/_config-child-theme-deep.scss]: There's another issue regarding this: #2282 - questioning whether this should be set at all.

Comment (scruffian) [reply rockfield/sass/_config-child-theme-deep.scss]: Yeah that's what I'm wondering. It feel unnecessary. Can we just remove it?

Comment (danieldudzic) [reply rockfield/sass/_config-child-theme-deep.scss]: We can remove it, but it seems that Rockfield/Varia will, regardless, compile and output code related to this:
```
ul,
ol {
	font-family: serif;
	font-family: var(--font-base, serif);
	margin: 0;
	padding-left: 32px;
}

```

Comment (scruffian) [reply rockfield/sass/_config-child-theme-deep.scss]: Do you know where that first `font-family: serif;` comes from?

Comment (danieldudzic) [reply rockfield/sass/_config-child-theme-deep.scss]: Both font-family's come from https://github.com/Automattic/themes/blob/3fa9d68c06fb4d174aebe58f58952061839efbb2/varia/sass/blocks/list/_style.scss#L3

Comment (scruffian) [reply rockfield/sass/_config-child-theme-deep.scss]: Looks like its there for IE11 support. Does this solution solve the original issue?

Comment (danieldudzic) [reply rockfield/sass/_config-child-theme-deep.scss]: Both options seem to be working fine: changing the family from `primary` to `secondary` or removing the code altogether.

Comment (scruffian) [reply rockfield/sass/_config-child-theme-deep.scss]: In that case let's remove the code. Less code = less bugs!

**Review by scruffian — APPROVED** [review]
LGTM

**Pull request merged**